src/settings/appsettings.cpp

```diff
@@ -664,6 +664,53 @@ QOnlineTts::Emotion AppSettings::defaultEmotion(QOnlineTranslator::Engine engine
     }
 }
 
+QMap<QOnlineTranslator::Language, QLocale::Country> AppSettings::regions(QOnlineTranslator::Engine engine) const
+{
+    switch (engine) {
+    case QOnlineTranslator::Google: {
+        QMap<QOnlineTranslator::Language, QLocale::Country> regions;
+        for (int i = 1; i <= QOnlineTranslator::Zulu; ++i)
+            if (!QOnlineTts::validRegions(static_cast<QOnlineTranslator::Language>(i)).isEmpty())
```

Comment:
Just iterate over regions directly using `regions`.

Reply:
Oh, yes! Or it will store loads of `AnyCountry` into config, which makes no sense.

Reply:
Well, it seems that in `QMap<QOnlineTranslator::Language, QLocale::Country> AppSettings::regions(QOnlineTranslator::Engine engine) const` I have no access to a `QOnlineTts` instance to call `const QMap<QOnlineTranslator::Language, QLocale::Country> &QOnlineTts::regions() const`?

Reply:
Oh, sorry, my bad, I for some reason thought that `regions` returns a map :smile: 
But what if `validRegions` will return a map? We could make a const static map in memory and just return it instead. Lookups will be much more efficient and we will be able to iterate over languages and their regions almost without cost.

Reply:
Did you mean it will return `const QMap<QOnlineTranslator::Language, QList<QLocale::Country>> &`? I think it's very nice to have. So should I open a new PR in the library repo (and remove the redundant `QOnlineTts::regionName` along the way)? The function is used three times in Crow, so it should be easy to modify.

Reply:
>Did you mean it will return const QMap<QOnlineTranslator::Language, QList<QLocale::Country>> &? I think it's very nice to have. 

Yep!

> So should I open a new PR in the library repo (and remove the redundant QOnlineTts::regionName along the way)?

It would be great!